Add action

nurmukhametov opened the pull request:
This PR contains a reworked version of the action from PR #1. There are two main differences: the use of `actions/tool-cache` and deriving default values for `architecture`/`platform`.

@ScatteredRay, thank you for your contribution. Your work was incredibly valuable and served as a solid foundation for this PR. It provided the necessary insights and direction, making the reworking process much smoother and more efficient.

This PR also includes a `README.md` derived from the draft version proposed by @maedoc.

Additionally, `DEVELOPMENT.md` and `.gitignore` were added to simplify the development process.

ScatteredRay committed:
Add action.yml, package.json

ScatteredRay committed:
Add src/main.js

nurmukhametov committed:
Use actions/tool-cache to download and extract

nurmukhametov committed:
Add package-lock.json

nurmukhametov committed:
Update README.md

nurmukhametov committed:
Add DEVELOPMENT.md

nurmukhametov committed:
Add bundled action under build/main.cjs

nurmukhametov committed:
Add .gitignore

aneshlya commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,28 @@
+# Install Dependencies
```
Please include prerequisites (e.g., Node.js version) before the installation steps.

nurmukhametov replied:
I have included a sentence that Node.js required, although I am not quite sure that it is worth mentioning any version there. My setup uses v22.2.0 whereas, e.g., `ubuntu-latest` uses v18.* and it looks fine.

aneshlya commented on README.md:
```diff
@@ -1,2 +1,71 @@
-# install-ispc-action
-Github Action to install ISPC compiler
+# Install ISPC GitHub Action
```
Include build status badges at the top of the README for better visibility.

nurmukhametov replied:
Done

aneshlya commented on src/main.js:
```diff
@@ -0,0 +1,165 @@
+    // Note: The GitHub API is rate limited to 60 requests per hour for unauthenticated users.
+    // I am not quite sure that it is reliable to query the latest version from
+    // the GitHub API because of the rate limits.
+    let response = await fetch(`https://api.github.com/repos/ispc/ispc/releases/latest`);
```
I don't know Javascript but it seems that there is `let` and `const` keywords (both can be used to declare variables in JavaScript). 
`let`: Allows you to reassign the variable after its initial declaration.
`const`: Does not allow reassignment. Once a value is assigned, it cannot be changed.

In this script most of the variables are `const`.

nurmukhametov replied:
It looks like the linter fixed it.

aneshlya commented on src/main.js:
```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache-2.0
```
The formatting of this file is strange (no space between `if` and `(`), space between `Error (``. Hard ro read. Could you use some JS beautifier? Like https://beautifier.io/

nurmukhametov replied:
I have added `run lint` and `run lint-fix` commands to invoke linter. Surprisingly enough, it also fixes `let/const` mess in variable declarations.

aneshlya reviewed (approved):
Looks good! I just have a couple of minor style comments.

nurmukhametov committed:
Add fallback for version detection logic

It may happen that runner is banned by API rate limits. In this case,
just clone git repository and fetch the latest tag to define the latest
version.

aneshlya commented on src/main.js:
```diff
@@ -0,0 +1,186 @@
+            execSync(`git -C ${repoPath} fetch --tags`);
+
+            let latestTagSHA = execSync(`git -C ${repoPath} rev-list --tags --max-count=1`).toString().trim();
+            version = execSync(`git -C ${repoPath} describe --tags ${latestTagSHA}`).toString().trim();
```
Should we remove the repo after determining the version?

nurmukhametov replied:
Fixed

nurmukhametov committed:
Add lint run command

Run npm run lint to check if src/main.js linted correctly.
Run npm run lint-fix to fix automatically.

nurmukhametov committed:
Lint code

nurmukhametov committed:
test.yml: check that required version is installed

aneshlya reviewed (approved):
LGTM!
